### DISCUSSION TITLE
Install latest podman version from rpm package

### DISCUSSION THREAD

**Discussion by b3n3w** [containers/podman#14891] [General]
Hi there,

i would like to install the latest podman version on our almalinux 9 test server. When trying to install the downloaded package from pkgs.org, it reports that the podman-catatonit should be installed first. But when trying to install this package, it claims also that podman 4.1.1 should be installed before.

How can i avoid this ?

`sudo rpm -i podman-4.1.1-1.el9.x86_64.rpm 
warning: podman-4.1.1-1.el9.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID 8483c65d: NOKEY
error: Failed dependencies:
	podman-catatonit >= 2:4.1.1-1.el9 is needed by podman-2:4.1.1-1.el9.x86_64
`

`sudo rpm -i podman-catatonit-4.1.1-1.el9.x86_64.rpm 
warning: podman-catatonit-4.1.1-1.el9.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID 8483c65d: NOKEY
error: Failed dependencies:
	podman = 2:4.1.1-1.el9 is needed by podman-catatonit-2:4.1.1-1.el9.x86_64
`

podman rpm: 
https://centos.pkgs.org/9-stream/centos-appstream-x86_64/podman-4.1.1-1.el9.x86_64.rpm.html

**Comment by vrothberg** [comment]
Thanks for reaching out, @b3n3w!

@jnovy @lsm5, do you know what's going on?

**Comment by b3n3w** [comment]
I "fixed" it by just installing the packages at the same time:

`sudo rpm -ivh podman-catatonit-4.1.1-1.el9.x86_64.rpm podman-4.1.1-1.el9.x86_64.rpm`

**Comment by lsm5** [comment]
@b3n3w yes, installing them together is the way to go. Also, an alternative to downloading + `rpm -ivh` would be to `sudo dnf install $PODMAN_RPM_URL $CATATONIT_RPM_URL`

**Reply by mailinglists35** [reply]
where to get `$PODMAN_RPM_URL` & `$CATATONIT_RPM_URL` for EL9/OL9 from? or did you simply meant passing the koji url directly to dnf?

**Comment by nikhilkd1999** [comment]
Hey @vrothberg / @b3n3w, I tried updating podman with cmd 

`rpm -U podman-4.9.4-0.1.module_el8+971+3d3df00d.x86_64.rpm podman-catatonit-4.9.4-0.1.module_el8+971+3d3df00d.x86_64.rpm ` in my RHEL 8 machine.

I used RPMs from [here](https://rpmfind.net/linux/rpm2html/search.php?query=podman%28x86-64%29&submit=Search+...&system=&arch=).

My existing Podman version was `4.6.1` and it got upgraded to `4.9.4-dev`.

Any idea why the `dev` version is getting installed? How do I get rid of `dev` version?

**Reply by lsm5** [reply]
we have built 4.9.4-1 for centos 8 stream but that hasn't landed yet in the main repos. You can get the rpms from: https://kojihub.stream.centos.org/koji/buildinfo?buildID=60260 . 

Also FYI, centos 8 stream will be frozen at the end of May 2024, so there won't be any further updates to podman on that environment  after that. You should consider switching to an EL 9 environment or look at getting your podman builds some other way.

**Reply by nikhilkd1999** [reply]
Thanks @lsm5!!

**Reply by mailinglists35** [reply]
@lsm5 what's the best strategy to override EL9 distro packages with those from latest build? manually download the corresponding rpm from koji and dnf localinstall / rpm -U ?

````
$ rpm -aq|grep podman
podman-5.2.2-15.0.1.el9_5.x86_64
cockpit-podman-93.1-1.el9_5.noarch
podman-plugins-5.2.2-15.0.1.el9_5.x86_64
podman-compose-1.0.6-3.el9.noarch
podman-docker-5.2.2-15.0.1.el9_5.noarch
````

**Reply by lsm5** [reply]
depends. if those are from a dnf repo, you could enable it and just run `dnf update`. Else you could download the rpms and run `dnf update $PATH_TO_RPM_FILE1 $PATH_TO_RPM_FILE2` . If it complains about conflicts you can try adding the options `--best` and/or `--allowerasing` .

**Comment by lsm5** [comment]
Don't think this needs to stay open. Please open a new issue / bugzilla if you notice issues with rpms.
